spec/requests/posts_controller_spec.rb

```diff
@@ -423,6 +423,62 @@
         end
       end
     end
+
+    context "api" do
+      let(:api_key) { user.generate_api_key(user) }
+      let(:master_key) { ApiKey.create_master_key }
+
+      # choosing an arbitrarily easy to mock trusted activity
+      it 'allows users with api key to bookmark posts' do
+        put "/posts/#{post.id}/bookmark.json", params: {
+          bookmarked: "true",
+          api_key: api_key.key
+        }
+
+        expect(response).to be_success
+        expect(PostAction.where(post: post, user: user).count).to eq(1)
```

Comment:
We need to test for the type of `PostAction` here. Otherwise, a flag post action being incorrectly created would also allow this test to pass.